### PR TITLE
MED-94 Availability UI

### PR DESCRIPTION
**Pull request by skazzp**
**Add availability page UI**
General view
image
With time slots
image
With options to edit/remove time slots, with notification to check overlays in time
image

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

### CODE REVIEW

File: apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import moment from 'moment';
+import { toast } from 'react-toastify';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+
+import 'react-big-calendar/lib/css/react-big-calendar.css';
+import { hoursSchema } from 'validation/selectHourRangeSchema';
+import { toastConfig } from 'utils/toastConfig';
+
+interface SelectHours {
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx]: create type file and put there

File: apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  start: number;
+  end: number;
+}
+
+export function useCalendar() {
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx]: i believe in terms of hooks, or you put in the hook file that Oleksi created or rename this file to hooks.tsx and use it here, from what i remember of the structure

File: apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx (hunk 1 above)
Comment (skazzp) [reply apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx]: done

File: apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx (hunk 2 above)
Comment (skazzp) [reply apps/medon-fe/src/components/AvailabilityCalendar/useCalendar.tsx]: renamed

**Review by Mati-Pereira — APPROVED** [review]

File: apps/medon-fe/src/components/AvailabilityCalendar/hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    setTimeSlots((prev) => [...prev.filter((e, i) => i !== editIndex)]);
+    setSelectedDay(undefined);
+    setEditIndex(null);
+  };
+
+  const handleSubmitEvent = (data: SelectHours) => {
+    const title = `${dayjs()
+      .hour(data.start)
+      .minute(0)
+      .format('HH:mm')} - ${dayjs().hour(data.end).minute(0).format('HH:mm')}`;
```
Comment (lyudad) [inline apps/medon-fe/src/components/AvailabilityCalendar/hooks.tsx]: format fo constant, it should be general for full project

Comment (skazzp) [reply apps/medon-fe/src/components/AvailabilityCalendar/hooks.tsx]: moved to constant

**Pull request merged**